addons/test_mail_full/tests/test_rating.py

```diff
@@ -20,38 +20,43 @@ def setUpClass(cls):
             'name': 'Test Rating',
             'user_id': cls.user_admin.id,
         })
+        cls.record_rating_without_mixin = cls.env['mail.test.rating.without.mixin'].create({
+            'partner_id': cls.partner_1.id,
+            'name': 'Test rating without rating mixin',
+            'user_id': cls.user_admin.id,
+        })
 
 
 @tagged('rating')
 class TestRatingFlow(TestRatingCommon):
 
     def test_initial_values(self):
-        record_rating = self.record_rating.with_env(self.env)
-        self.assertFalse(record_rating.rating_ids)
-        self.assertEqual(record_rating.message_partner_ids, self.partner_admin)
-        self.assertEqual(len(record_rating.message_ids), 1)
+        for record_rating in [self.record_rating, self.record_rating_without_mixin]:
+            record_rating = record_rating.with_env(self.env)
+            self.assertFalse(record_rating.rating_ids)
+            self.assertEqual(record_rating.message_partner_ids, self.partner_admin)
+            self.assertEqual(len(record_rating.message_ids), 1)
 
     @users('employee')
     @mute_logger('odoo.addons.mail.models.mail_mail')
     def test_rating_prepare(self):
-        record_rating = self.record_rating.with_env(self.env)
-
-        # prepare rating token
-        access_token = record_rating._rating_get_access_token()
-
-        # check rating creation
-        rating = record_rating.rating_ids
-        self.assertEqual(rating.access_token, access_token)
-        self.assertFalse(rating.consumed)
-        self.assertFalse(rating.is_internal)
-        self.assertEqual(rating.partner_id, self.partner_1)
-        self.assertEqual(rating.rated_partner_id, self.user_admin.partner_id)
-        self.assertFalse(rating.rating)
-
-    @users('employee')
-    @mute_logger('odoo.addons.mail.models.mail_mail')
-    def test_rating_rating_apply(self):
-        record_rating = self.record_rating.with_env(self.env)
+        for record_rating in [self.record_rating, self.record_rating_without_mixin]:
+            record_rating = record_rating.with_env(self.env)
+
+            # prepare rating token
+            access_token = record_rating._rating_get_access_token()
+
+            # check rating creation
+            rating = record_rating.rating_ids
+            self.assertEqual(rating.access_token, access_token)
+            self.assertFalse(rating.consumed)
+            self.assertFalse(rating.is_internal)
+            self.assertEqual(rating.partner_id, self.partner_1)
+            self.assertEqual(rating.rated_partner_id, self.user_admin.partner_id)
+            self.assertFalse(rating.rating)
+
+    def _test_rating_rating_apply(self, record_rating, expected_subtype, is_rating_mixin_test):
```

Comment:
Seems it is a good candidate to start using subTests (see https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests ) :) would lead to clearer test writing.

Reply:
Thanks for the tip. I have used subTest when possible. 
I think it is not possible for the performance test as the parametrize decorator (as in pytest) is missing and we need the warmup decorator so for those test, I have kept the private method.

Note that one drawback of subtest is that it introduces 2 more indentations.